### PR TITLE
Apply consistent set of attributes to metrics and stacks sent in logs

### PR DESCRIPTION
**Pull request by lachmatt**
## Why

Have consistent set of attributes across different signals.
## What

- add attributes commonly added by Otel-based instrumentations to metrics/stacks sent in logs
- rename `SIGNALFX_TRACE_GLOBAL_TAGS` to `SIGNALFX_GLOBAL_TAGS` as it applies broader (spans and stack sent in logs before, now also to metrics)
- replace `Verify` usage with standard assertions for simple scenario in `AlwaysOnProfilerTests`
- replace `Xunit` assertions with `FluentAssertions` in `RuntimeMetricsTests`

## Tests

Included in PR.

### CODE REVIEW

**Review by Kielek — APPROVED** [review]

**Comment by theletterf** [comment]
@lachmatt Can this be documented without documenting the runtime / profiling bits?

File: docs/advanced-config.md [hunk 1]
```diff
@@ -66,7 +66,8 @@ The following settings are common to most instrumentation scenarios:
 |-|-|-|
 | `SIGNALFX_DISABLED_INTEGRATIONS` | Comma-separated list of disabled library instrumentations. The available integration ID values can be found [here](instrumented-libraries.md). |  |
 | `SIGNALFX_RECORDED_VALUE_MAX_LENGTH` | The maximum length an attribute value can have. Values longer than this are truncated. Values are discarded entirely when set to 0, and ignored when set to a negative value. | `12000` |
-| `SIGNALFX_TRACE_GLOBAL_TAGS` | Comma-separated list of key-value pairs that specify global span tags. For example: `"key1:val1,key2:val2"` |  |
+| `SIGNALFX_TRACE_GLOBAL_TAGS` | Deprecated. Use `SIGNALFX_GLOBAL_TAGS` instead. |  |
+| `SIGNALFX_GLOBAL_TAGS` | Comma-separated list of key-value pairs that specify global tags added to all telemetry signals. For example: `"key1:val1,key2:val2"`. If not set, value of `SIGNALFX_TRACE_GLOBAL_TAGS` is used if set. |  |
```
Comment (theletterf) [inline docs/advanced-config.md:70]:
```suggestion
| `SIGNALFX_GLOBAL_TAGS` | Comma-separated list of key-value pairs that specify global tags added to all telemetry signals. For example: `"key1:val1,key2:val2"`. If unset, the value of the `SIGNALFX_TRACE_GLOBAL_TAGS` environment variable is used, if available. |  |
```

What happens if `SIGNALFX_TRACE_GLOBAL_TAGS` is also unset?

**Review by theletterf — APPROVED** [review]

File: tracer/src/Datadog.Trace/Configuration/TracerSettings.cs [hunk 2]
```diff
@@ -94,6 +94,7 @@ public TracerSettings(IConfigurationSource source)
                                           100;
 
             GlobalTags = source?.GetDictionary(ConfigurationKeys.GlobalTags) ??
+                         source?.GetDictionary("SIGNALFX_TRACE_GLOBAL_TAGS") ??
```
Comment (pjanotti) [inline tracer/src/Datadog.Trace/Configuration/TracerSettings.cs:97]: nit: add a comment about this still be here to help transition to the new env var

Comment (pjanotti) [reply tracer/src/Datadog.Trace/Configuration/TracerSettings.cs:97]: On the file below we have: `// backwards compatibility for names used in the past`

**Review by pjanotti — APPROVED** [review]
LGTM. A tiny suggestion about adding a comment.

File: tracer/src/Datadog.Trace/Configuration/TracerSettings.cs (hunk 2 above)
Comment (lachmatt) [inline tracer/src/Datadog.Trace/Configuration/TracerSettings.cs:97]:
```suggestion
                         // backwards compatibility for names used in the past
                         source?.GetDictionary("SIGNALFX_TRACE_GLOBAL_TAGS") ??
```

Comment (lachmatt) [reply tracer/src/Datadog.Trace/Configuration/TracerSettings.cs:98]: added in https://github.com/signalfx/signalfx-dotnet-tracing/pull/687/commits/bac99447db1db4d758c6faab0f06f9eee4853042

**Pull request merged**